### PR TITLE
Fix to show error state rather than spinner

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
## Fixes 
Fixes [HAC-3050](https://issues.redhat.com/browse/HAC-3050)

## Description
Handle errors at top level pages and display an error state if an error is returned while loading the data.
Created an ErrorState component that handles showing errors consistently, and generically handles 404 errors.

## Type of change
- [x] Bugfix

## Screen shots / Gifs for design review 

image

/cc @MariaLeonova

### CODE REVIEW

**Comment by MariaLeonova** [comment]
Thank you, @jeff-phillips-18 ! This looks great!

File: config/remotePlugin.js [hunk 1]
```diff
@@ -349,6 +349,32 @@ const routeExtensions = [
...
+      path: '/stonesoup',
+      exact: false,
+      component: {
+        $codeRef: 'NotFound',
+      },
+    },
+    flags: {
+      required: ['SIGNUP'],
+    },
+  },
```
Comment (rohitkrai03) [inline config/remotePlugin.js:364]: We don't need to add the deprecated `console` extensions anymore.

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

File: config/remotePlugin.js (hunk 1 above)
Comment (jeff-phillips-18) [reply config/remotePlugin.js:364]: Shall I remove the others?

Comment (christianvogt) [reply config/remotePlugin.js:364]: @jeff-phillips-18 may as well if they are redundant and unnecessary.

**Pull request merged**